Solutions/Corvus.Monitoring.AspnetCore.Mvc.Demo/Program.cs

```diff
@@ -0,0 +1,41 @@
+// <copyright file="Program.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddControllersWithViews();
+
+// Add Corvus.Monitoring. This will add the "null" implementations of IOperationsInstrumentation and IExceptionsInstrumentation
+// by default.
+builder.Services.AddInstrumentation();
+
+WebApplication app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+
+    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+app.MapControllerRoute(
+    name: "default",
+    pattern: "{controller=Home}/{action=Index}/{id?}");
+
+app.Run();
+
+/// <summary>
+/// Partial program class to make the actual one public so it can be accessed for testing purposes.
```

Comment:
I was unaware you could do this hack.

I can't help thinking, though, that if we positively need this to export a type called `Program` with a method called `Main`, we should probably just do it the old-fashioned way?

Or is the goal here that the source file mostly resembles a normal-looking ASP.NET Core project?

Reply:
I agree it looks hacky. However, it's the approach described in the Microsoft docs - https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-7.0, so I think it makes sense to leave it like this.